.github/workflows/scheduled-ansible-pull.yml

```diff
@@ -0,0 +1,70 @@
+name: Check if the collection is up to date with the ansible role
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 4 * * *'
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    outputs:
+      check-output: ${{ steps.check-diff.outputs.sync-needed }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@master
+        with:
+          repository: DataDog/ansible-datadog
+          path: ansible-role
+      - name: Check diff between roles/agent and ansible-role main branch
+        id: check-diff
+        run: |
+          set 
+          diff -x "\.*" -x CHANGELOG.md -x CONTRIBUTING.md -x LICENSE -x ci_test -x manual_tests -x "requirements*" -pur roles/agent/ ansible-role/ || echo "sync-needed=true" >> $GITHUB_OUTPUT
+          echo "DEBUG: $GITHUB_OUTPUT"
+
+  execute:
+    runs-on: ubuntu-latest
+    needs: check
+    if: "${{ needs.check.outputs.check-output == 'true' }}"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@master
+        with:
+          repository: DataDog/ansible-datadog
+          path: roles/agent
+      - name: Apply Patches if there is any
+        run: |
+          for patch in $(ls patches)
+          do
+            echo "${patch}..."
+            git -C roles/agent/ apply --verbose "../../patches/$patch"
+          done
```

Comment:
It seems the patches are expected to be in the git format, so you could simplify this with `git -C roles/agent am patches/*.patch` no?

Reply:
hmm should work, wanted to have a small `"${patch}..."` before but I guess it's not needed

Reply:
actually wont work because `patches/*.patch` is going to be expanded to `patches/somefile.patch` and `git -C roles/agent am patches/somefile.patch` doesn't work because of `-C`

Reply:
Ah? Oh well it's not a big deal anyway